src/App/Fossa/Analyze.hs

```diff
@@ -257,12 +267,18 @@ analyze (BaseDir basedir) destination override unpackArchives jsonOutput enableV
   let filteredProjects = filterProjects (BaseDir basedir) projectResults
 
   -- Need to check if vendored is empty as well, even if its a boolean that vendoredDeps exist
-  case checkForEmptyUpload projectResults filteredProjects [manualSrcUnits, vsiResults] of
+  uploadLocator <- case checkForEmptyUpload projectResults filteredProjects [manualSrcUnits, vsiResults] of
     NoneDiscovered -> Diag.fatal ErrNoProjectsDiscovered
     FilteredAll count -> Diag.fatal (ErrFilteredAllProjects count projectResults)
     FoundSome sourceUnits -> case destination of
-      OutputStdout -> logStdout . decodeUtf8 . Aeson.encode $ buildResult manualSrcUnits filteredProjects
-      UploadScan opts metadata -> uploadSuccessfulAnalysis (BaseDir basedir) opts metadata jsonOutput override sourceUnits
+      OutputStdout -> do
+        _ <- logStdout . decodeUtf8 . Aeson.encode $ buildResult manualSrcUnits filteredProjects
+        pure Nothing
+      UploadScan opts metadata -> do
+        locator <- uploadSuccessfulAnalysis (BaseDir basedir) opts metadata jsonOutput override sourceUnits
+        pure $ Just locator
+
+  doAssertRevisionBinaries iatAssertion apiOpts uploadLocator
```

Comment:
[nit] let's move this up into the `UploadScan` case branch above -- prevents us from having to pass out an `uploadLocator`, which only happens in that branch anyway

in that branch we have the concrete `opts` and `locator` too, so we can remove the `Maybe` wrappers from the args to `doAssertRevisionBinaries`